client/trino-jdbc/src/main/java/io/trino/jdbc/TrinoDataSource.java

```diff
@@ -0,0 +1,305 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.jdbc;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.trino.client.uri.PropertyName;
+
+import javax.sql.DataSource;
+
+import java.io.Closeable;
+import java.io.PrintWriter;
+import java.sql.Connection;
+import java.sql.Driver;
+import java.sql.SQLException;
+import java.util.Properties;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+import java.util.logging.Logger;
+
+import static io.trino.client.uri.PropertyName.CATALOG;
+import static io.trino.client.uri.PropertyName.PASSWORD;
+import static io.trino.client.uri.PropertyName.SCHEMA;
+import static io.trino.client.uri.PropertyName.SESSION_USER;
+import static io.trino.client.uri.PropertyName.SQL_PATH;
+import static io.trino.client.uri.PropertyName.USER;
+import static java.lang.String.format;
+
+public class TrinoDataSource
+        implements DataSource
+{
+    private String url;
+    private PrintWriter printWriter;
+
+    private final Properties connectionProperties = new Properties();
+    private final TrinoDriver trinoDriver;
+
+    private ReentrantReadWriteLock lock = new ReentrantReadWriteLock();
+
+    public TrinoDataSource()
+    {
+        trinoDriver = new TrinoDriver();
```

Comment:
Shouldn't we use a NonRegisteringDriver here ?

From the JavaDocs 

```
A driver that is accessed via a DataSource object does not register itself with the DriverManager. Rather, a DataSource object is retrieved through a lookup operation and then used to create a Connection object. With a basic implementation, the connection obtained through a DataSource object is identical to a connection obtained through the DriverManager facility.
```

Reply:
Good catch, changed.